client/rest-high-level/src/test/java/org/elasticsearch/client/security/PutRoleMappingRequestTests.java

```diff
@@ -0,0 +1,146 @@
+
+        final PutRoleMappingRequest putRoleMappingRequest = new PutRoleMappingRequest(name, enabled, roles, rules, metadata, refreshPolicy);
+
+        final XContentBuilder builder = XContentFactory.jsonBuilder();
+        putRoleMappingRequest.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        final String output = Strings.toString(builder);
+        final String expected =
+             "{"+
+               "\"enabled\":" + enabled + "," +
+               "\"roles\":[\"superuser\"]," +
+               "\"rules\":{" +
+                   "\"field\":{\"username\":[\"user\"]}" +
+               "}," +
+               "\"metadata\":{\"k1\":\"v1\"}" +
+             "}";
+
+        assertThat(output, equalTo(expected));
+    }
+
+    public void testEqualsHashCode() {
+        final String name = randomAlphaOfLength(5);
+        final boolean enabled = randomBoolean();
+        final List<String> roles = Collections.singletonList("superuser");
+        final RoleMapperExpression rules = FieldRoleMapperExpression.ofUsername("user");
+        final Map<String, Object> metadata = new HashMap<>();
+        metadata.put("k1", "v1");
+        final RefreshPolicy refreshPolicy = randomFrom(RefreshPolicy.values());
+
+        PutRoleMappingRequest putRoleMappingRequest = new PutRoleMappingRequest(name, enabled, roles, rules, metadata, refreshPolicy);
+        assertNotNull(putRoleMappingRequest);
+
+        EqualsHashCodeTestUtils.checkEqualsAndHashCode(putRoleMappingRequest, (original) -> {
+            return new PutRoleMappingRequest(original.getName(), original.isEnabled(), original.getRoles(), original.getRules(), original
+                    .getMetadata(), original.getRefreshPolicy());
+        });
+        EqualsHashCodeTestUtils.checkEqualsAndHashCode(putRoleMappingRequest, (original) -> {
+            return new PutRoleMappingRequest(original.getName(), original.isEnabled(), original.getRoles(), original.getRules(), original
+                    .getMetadata(), original.getRefreshPolicy());
+        }, PutRoleMappingRequestTests::mutateTestItem);
+    }
+
+    private static PutRoleMappingRequest mutateTestItem(PutRoleMappingRequest original) {
+        switch (randomIntBetween(0, 2)) {
+        case 0:
+            return new PutRoleMappingRequest(randomAlphaOfLength(5), original.isEnabled(), original.getRoles(), original.getRules(),
+                    original.getMetadata(), original.getRefreshPolicy());
+        case 1:
+            return new PutRoleMappingRequest(original.getName(), !original.isEnabled(), original.getRoles(), original.getRules(), original
+                    .getMetadata(), original.getRefreshPolicy());
+        case 2:
+            return new PutRoleMappingRequest(original.getName(), original.isEnabled(), original.getRoles(), FieldRoleMapperExpression
+                    .ofGroups("group"), original.getMetadata(), original.getRefreshPolicy());
+        default:
```

Comment:
Since this can never be anything else than `0,1,2`, maybe make the default case one of the permutations?